lib/s3middleware/tagging.ts

```diff
@@ -100,7 +114,46 @@ function _validateTags(tags: Array<{ Key: string[], Value: string[] }>) {
     }
     // not repeating keys
     if (tags.length > Object.keys(tagsResult).length) {
-        return errors.InvalidTag.customizeDescription('Cannot provide ' +
+        return errors.InvalidTag.customizeDescription(
+            'Cannot provide multiple Tags with the same key'
+        );
+    }
+    return tagsResult;
+}
+
+/** validateTags - Validate tags, throwing an error if tags are invalid
+* @param tags - tags parsed from xml to be validated
+* @param tags[].Key - Name of the tag
+* @param tags[].Value - Value of the tag
+* @return tagsResult - return object tags on success
+* { key: value }; error on failure
+*/
+export function validateTags(tags: Array<BucketTag>) {
```

Comment:
Ok after looking again I have two comments on this function.
1. Why do we return the tags back? If the tags are valid then the input is the same as the output, if they are not valid then its an error response. Why not just rename the function to `areTagsValid` and return a bool or false? 
2. Or if you want an explicit error keep it validate tags but return `error | undefined` instead.

Reply:
I agree with approach 1, the tags are also validated in the APIs from the provided XML, so throwing an error here is not required. Done in https://github.com/scality/Arsenal/pull/1933/commits/9c5365cca8d2dfc6445792f2b23aa14586bd8cb4#diff-7aa7eeaed88643ea97d57a5f15ddc2e64c380190641c53bb0c3f64d67807a475R128